packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -246,30 +254,61 @@ class _PosixUtils extends OperatingSystemUtils {
     return _fileSystem.file(path);
   }
 
+  @override
+  String get pathVarSeparator => ':';
+}
+
+class MacOSUtils extends _PosixUtils {
+  MacOSUtils({
+    @required FileSystem fileSystem,
+    @required Logger logger,
+    @required Platform platform,
+    @required ProcessManager processManager,
+  }) : super(
+          fileSystem: fileSystem,
+          logger: logger,
+          platform: platform,
+          processManager: processManager,
+        );
+
   String _name;
 
   @override
   String get name {
     if (_name == null) {
-      if (_platform.isMacOS) {
-        final List<RunResult> results = <RunResult>[
-          _processUtils.runSync(<String>['sw_vers', '-productName']),
-          _processUtils.runSync(<String>['sw_vers', '-productVersion']),
-          _processUtils.runSync(<String>['sw_vers', '-buildVersion']),
-          _processUtils.runSync(<String>['uname', '-m']),
-        ];
-        if (results.every((RunResult result) => result.exitCode == 0)) {
-          _name = '${results[0].stdout.trim()} ${results[1].stdout
-              .trim()} ${results[2].stdout.trim()} ${results[3].stdout.trim()}';
-        }
+      final List<RunResult> results = <RunResult>[
+        _processUtils.runSync(<String>['sw_vers', '-productName']),
+        _processUtils.runSync(<String>['sw_vers', '-productVersion']),
+        _processUtils.runSync(<String>['sw_vers', '-buildVersion']),
+      ];
+      if (results.every((RunResult result) => result.exitCode == 0)) {
+        _name =
+            '${results[0].stdout.trim()} ${results[1].stdout.trim()} ${results[2].stdout.trim()} ${getNameForDarwinArch(hardwareArchitecture)}';
       }
       _name ??= super.name;
     }
     return _name;
   }
 
-  @override
-  String get pathVarSeparator => ':';
+  DarwinArch _hardwareArchitecture;
+
+  /// Physical underlying architecture.
+  ///
+  /// On ARM return arm64, even when this process is running in Rosetta.
+  DarwinArch get hardwareArchitecture {
```

Comment:
Per your suggestion @jonahwilliams I'm going to think one step ahead--this will be needed outside just the name to detect how certain commands will need to be run.

Created a new `MacOSUtils` (not `_MacOSUtils` since I'll want to grab the arch from other places) and expose the arch.

WDYT?

Reply:
It would be nice to not need to know the specific subtype of os utils in order to call this function.

Instead of exposing a DarwinArch (since that is really a target architecture, we can't target 32bit arm), you could add darwin_arm64 something to the HostPlatform  (https://github.com/flutter/flutter/blob/345188d40e576ffb6c97fee60446af58b8c9965a/packages/flutter_tools/lib/src/build_info.dart#L399) And then have a getter on os utils to return the current host platform. If we eventually support linux arm and windows arm this will come in handy too.

You could then augment that with a helper/extension for `isArm/isX86`

Reply:
> Instead of exposing a DarwinArch (since that is really a target architecture, we can't target 32bit arm), you could add darwin_arm64 something to the HostPlatform

That's way better.